packages/library-base/src/commonMain/kotlin/io/realm/kotlin/ext/RealmAnyExt.kt

```diff
@@ -11,3 +16,69 @@ import io.realm.kotlin.types.RealmObject
  */
 public inline fun <reified T : BaseRealmObject> RealmAny.asRealmObject(): T =
     asRealmObject(T::class)
+
+/**
+ * Create a [RealmAny] encapsulating the [value] argument.
+ *
+ * This corresponds to calling [RealmAny.create]-variant with the specific typed non-null argument.
+ *
+ * @param value the value that should be wrapped in a [RealmAny].
+ * @return a [RealmAny] wrapping the [value] argument, or `null` if [value] is null.
+ */
+@Suppress("ComplexMethod")
+public fun realmAnyOf(value: Any?): RealmAny? {
+    return when (value) {
+        (value == null) -> null
+        is Boolean -> RealmAny.create(value)
+        is Byte -> RealmAny.create(value)
+        is Char -> RealmAny.create(value)
+        is Short -> RealmAny.create(value)
+        is Int -> RealmAny.create(value)
+        is Long -> RealmAny.create(value)
+        is Float -> RealmAny.create(value)
+        is Double -> RealmAny.create(value)
+        is String -> RealmAny.create(value)
+        is Decimal128 -> RealmAny.create(value)
+        is ObjectId -> RealmAny.create(value)
+        is ByteArray -> RealmAny.create(value)
+        is RealmInstant -> RealmAny.create(value)
+        is RealmUUID -> RealmAny.create(value)
+        is RealmObject -> RealmAny.create(value)
+        is DynamicRealmObject -> RealmAny.create(value)
+        is Set<*> -> RealmAny.create(value.map { realmAnyOf(it) }.toRealmSet())
```

Comment:
Sets cannot contain other collections either so no reason to bother 😅 But for list it could make sense, but I wouldn't think that it would work when `realmAnyOf` is not the last call of the method 🤔